`$dbg` statement in `sdl/examples/basic_window` results in compile error

### Describe the bug

`$dbg` statement in `sdl/examples/basic_window` results in compile error

### Reproduction Steps

`basic_window.v` with the following contents:
```v
module main

import sdl

fn main() {
	sdl.init(sdl.init_video)
	window := sdl.create_window('Hello SDL2'.str, 300, 300, 500, 300, 0)
	renderer := sdl.create_renderer(window, -1, u32(sdl.RendererFlags.accelerated) | u32(sdl.RendererFlags.presentvsync))

	mut should_close := false
	for {
		evt := sdl.Event{}
		for 0 < sdl.poll_event(&evt) {
			match evt.@type {
				.quit { should_close = true }
				else {}
			}
		}
		if should_close {
			break
		}
    $dbg // <- adding this yields a compile error

		sdl.set_render_draw_color(renderer, 255, 55, 55, 255)
		sdl.render_clear(renderer)
		sdl.render_present(renderer)
	}

	sdl.destroy_renderer(renderer)
	sdl.destroy_window(window)
	sdl.quit()
}
```

### Expected Behavior

Clean compile + run - and the debugger invoked

### Current Behavior

```
  v run examples/basic_window/
==================
/dev/shm/v_1000/basic_window.01HNJT6T4C374YQ9WZ2C5DPYX7.tmp.c:23075: warning: assignment discards qualifiers from pointer target type
/dev/shm/v_1000/basic_window.01HNJT6T4C374YQ9WZ2C5DPYX7.tmp.c:31646: error: cannot convert 'struct SDL_Window **' to 'struct SDL_Window'
...
==================
(Use `v -cg` to print the entire error message)
```

### Possible Solution

???

### Additional Information/Context

_No response_

### V version

V 0.4.4 fb0efc0

### Environment details (OS name and version, etc.)

Linux Arch based (Endeavor OS)
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -141,63 +141,62 @@ mut:
 	inside_dump_fn            bool
 	expected_fixed_arr        bool
 	inside_for_c_stmt         bool
-	// inside_comptime_for_field bool
-	inside_cast_in_heap     int // inside cast to interface type in heap (resolve recursive calls)
-	inside_cast             bool
-	inside_const            bool
-	inside_array_item       bool
-	inside_const_opt_or_res bool
-	inside_lambda           bool
-	inside_cinit            bool
-	inside_interface_deref  bool
-	last_tmp_call_var       []string
-	loop_depth              int
-	ternary_names           map[string]string
-	ternary_level_names     map[string][]string
-	arraymap_set_pos        int      // map or array set value position
-	stmt_path_pos           []int    // positions of each statement start, for inserting C statements before the current statement
-	skip_stmt_pos           bool     // for handling if expressions + autofree (since both prepend C statements)
-	left_is_opt             bool     // left hand side on assignment is an option
-	right_is_opt            bool     // right hand side on assignment is an option
-	assign_ct_type          ast.Type // left hand side resolved comptime type
-	indent                  int
-	empty_line              bool
-	assign_op               token.Kind // *=, =, etc (for array_set)
-	defer_stmts             []ast.DeferStmt
-	defer_ifdef             string
-	defer_profile_code      string
-	defer_vars              []string
-	str_types               []StrType       // types that need automatic str() generation
-	generated_str_fns       []StrType       // types that already have a str() function
-	str_fn_names            []string        // remove duplicate function names
-	threaded_fns            shared []string // for generating unique wrapper types and fns for `go xxx()`
-	waiter_fns              shared []string // functions that wait for `go xxx()` to finish
-	needed_equality_fns     []ast.Type
-	generated_eq_fns        []ast.Type
-	array_sort_fn           shared []string
-	array_contains_types    []ast.Type
-	array_index_types       []ast.Type
-	auto_fn_definitions     []string // auto generated functions definition list
-	sumtype_casting_fns     []SumtypeCastingFn
-	anon_fn_definitions     []string     // anon generated functions definition list
-	sumtype_definitions     map[int]bool // `_TypeA_to_sumtype_TypeB()` fns that have been generated
-	trace_fn_definitions    []string
-	json_types              []ast.Type // to avoid json gen duplicates
-	pcs                     []ProfileCounterMeta // -prof profile counter fn_names => fn counter name
-	hotcode_fn_names        []string
-	hotcode_fpaths          []string
-	embedded_files          []ast.EmbeddedFile
-	sql_i                   int
-	sql_stmt_name           string
-	sql_bind_name           string
-	sql_idents              []string
-	sql_idents_types        []ast.Type
-	sql_left_type           ast.Type
-	sql_table_name          string
-	sql_fkey                string
-	sql_parent_id           string
-	sql_side                SqlExprSide // left or right, to distinguish idents in `name == name`
-	strs_to_free0           []string    // strings.Builder
+	inside_cast_in_heap       int // inside cast to interface type in heap (resolve recursive calls)
+	inside_cast               bool
+	inside_const              bool
+	inside_array_item         bool
+	inside_const_opt_or_res   bool
+	inside_lambda             bool
+	inside_cinit              bool
+	inside_interface_deref    bool
+	last_tmp_call_var         []string
+	loop_depth                int
+	ternary_names             map[string]string
+	ternary_level_names       map[string][]string
+	arraymap_set_pos          int      // map or array set value position
+	stmt_path_pos             []int    // positions of each statement start, for inserting C statements before the current statement
+	skip_stmt_pos             bool     // for handling if expressions + autofree (since both prepend C statements)
+	left_is_opt               bool     // left hand side on assignment is an option
+	right_is_opt              bool     // right hand side on assignment is an option
+	assign_ct_type            ast.Type // left hand side resolved comptime type
+	indent                    int
+	empty_line                bool
+	assign_op                 token.Kind // *=, =, etc (for array_set)
+	defer_stmts               []ast.DeferStmt
+	defer_ifdef               string
+	defer_profile_code        string
+	defer_vars                []string
+	str_types                 []StrType       // types that need automatic str() generation
+	generated_str_fns         []StrType       // types that already have a str() function
+	str_fn_names              []string        // remove duplicate function names
+	threaded_fns              shared []string // for generating unique wrapper types and fns for `go xxx()`
+	waiter_fns                shared []string // functions that wait for `go xxx()` to finish
+	needed_equality_fns       []ast.Type
+	generated_eq_fns          []ast.Type
+	array_sort_fn             shared []string
+	array_contains_types      []ast.Type
+	array_index_types         []ast.Type
+	auto_fn_definitions       []string // auto generated functions definition list
+	sumtype_casting_fns       []SumtypeCastingFn
+	anon_fn_definitions       []string     // anon generated functions definition list
+	sumtype_definitions       map[int]bool // `_TypeA_to_sumtype_TypeB()` fns that have been generated
+	trace_fn_definitions      []string
+	json_types                []ast.Type // to avoid json gen duplicates
+	pcs                       []ProfileCounterMeta // -prof profile counter fn_names => fn counter name
+	hotcode_fn_names          []string
+	hotcode_fpaths            []string
+	embedded_files            []ast.EmbeddedFile
+	sql_i                     int
+	sql_stmt_name             string
+	sql_bind_name             string
+	sql_idents                []string
+	sql_idents_types          []ast.Type
+	sql_left_type             ast.Type
+	sql_table_name            string
+	sql_fkey                  string
+	sql_parent_id             string
+	sql_side                  SqlExprSide // left or right, to distinguish idents in `name == name`
+	strs_to_free0             []string    // strings.Builder
 	// strs_to_free          []string // strings.Builder
 	// tmp_arg_vars_to_free  []string
 	// autofree_pregen       map[string]string
@@ -4022,6 +4021,8 @@ fn (mut g Gen) debugger_stmt(node ast.DebuggerStmt) {
 							''
 						} else if str_method_expects_ptr && !obj.typ.is_ptr() {
 							'&'
+						} else if !str_method_expects_ptr && obj.typ.is_ptr() {
+							'*'.repeat(obj.typ.nr_muls())
 						} else if obj.is_auto_heap && var_typ.is_ptr() && str_method_expects_ptr {
 							'*'
 						} else if !obj.is_auto_heap && var_typ.is_ptr() && str_method_expects_ptr {

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -141,62 +141,63 @@ mut:
 	inside_dump_fn            bool
 	expected_fixed_arr        bool
 	inside_for_c_stmt         bool
-	inside_cast_in_heap       int // inside cast to interface type in heap (resolve recursive calls)
-	inside_cast               bool
-	inside_const              bool
-	inside_array_item         bool
-	inside_const_opt_or_res   bool
-	inside_lambda             bool
-	inside_cinit              bool
-	inside_interface_deref    bool
-	last_tmp_call_var         []string
-	loop_depth                int
-	ternary_names             map[string]string
-	ternary_level_names       map[string][]string
-	arraymap_set_pos          int      // map or array set value position
-	stmt_path_pos             []int    // positions of each statement start, for inserting C statements before the current statement
-	skip_stmt_pos             bool     // for handling if expressions + autofree (since both prepend C statements)
-	left_is_opt               bool     // left hand side on assignment is an option
-	right_is_opt              bool     // right hand side on assignment is an option
-	assign_ct_type            ast.Type // left hand side resolved comptime type
-	indent                    int
-	empty_line                bool
-	assign_op                 token.Kind // *=, =, etc (for array_set)
-	defer_stmts               []ast.DeferStmt
-	defer_ifdef               string
-	defer_profile_code        string
-	defer_vars                []string
-	str_types                 []StrType       // types that need automatic str() generation
-	generated_str_fns         []StrType       // types that already have a str() function
-	str_fn_names              []string        // remove duplicate function names
-	threaded_fns              shared []string // for generating unique wrapper types and fns for `go xxx()`
-	waiter_fns                shared []string // functions that wait for `go xxx()` to finish
-	needed_equality_fns       []ast.Type
-	generated_eq_fns          []ast.Type
-	array_sort_fn             shared []string
-	array_contains_types      []ast.Type
-	array_index_types         []ast.Type
-	auto_fn_definitions       []string // auto generated functions definition list
-	sumtype_casting_fns       []SumtypeCastingFn
-	anon_fn_definitions       []string     // anon generated functions definition list
-	sumtype_definitions       map[int]bool // `_TypeA_to_sumtype_TypeB()` fns that have been generated
-	trace_fn_definitions      []string
-	json_types                []ast.Type // to avoid json gen duplicates
-	pcs                       []ProfileCounterMeta // -prof profile counter fn_names => fn counter name
-	hotcode_fn_names          []string
-	hotcode_fpaths            []string
-	embedded_files            []ast.EmbeddedFile
-	sql_i                     int
-	sql_stmt_name             string
-	sql_bind_name             string
-	sql_idents                []string
-	sql_idents_types          []ast.Type
-	sql_left_type             ast.Type
-	sql_table_name            string
-	sql_fkey                  string
-	sql_parent_id             string
-	sql_side                  SqlExprSide // left or right, to distinguish idents in `name == name`
-	strs_to_free0             []string    // strings.Builder
+	// inside_comptime_for_field bool
+	inside_cast_in_heap     int // inside cast to interface type in heap (resolve recursive calls)
+	inside_cast             bool
+	inside_const            bool
+	inside_array_item       bool
+	inside_const_opt_or_res bool
+	inside_lambda           bool
+	inside_cinit            bool
+	inside_interface_deref  bool
+	last_tmp_call_var       []string
+	loop_depth              int
+	ternary_names           map[string]string
+	ternary_level_names     map[string][]string
+	arraymap_set_pos        int      // map or array set value position
+	stmt_path_pos           []int    // positions of each statement start, for inserting C statements before the current statement
+	skip_stmt_pos           bool     // for handling if expressions + autofree (since both prepend C statements)
+	left_is_opt             bool     // left hand side on assignment is an option
+	right_is_opt            bool     // right hand side on assignment is an option
+	assign_ct_type          ast.Type // left hand side resolved comptime type
+	indent                  int
+	empty_line              bool
+	assign_op               token.Kind // *=, =, etc (for array_set)
+	defer_stmts             []ast.DeferStmt
+	defer_ifdef             string
+	defer_profile_code      string
+	defer_vars              []string
+	str_types               []StrType       // types that need automatic str() generation
+	generated_str_fns       []StrType       // types that already have a str() function
+	str_fn_names            []string        // remove duplicate function names
+	threaded_fns            shared []string // for generating unique wrapper types and fns for `go xxx()`
+	waiter_fns              shared []string // functions that wait for `go xxx()` to finish
+	needed_equality_fns     []ast.Type
+	generated_eq_fns        []ast.Type
+	array_sort_fn           shared []string
+	array_contains_types    []ast.Type
+	array_index_types       []ast.Type
+	auto_fn_definitions     []string // auto generated functions definition list
+	sumtype_casting_fns     []SumtypeCastingFn
+	anon_fn_definitions     []string     // anon generated functions definition list
+	sumtype_definitions     map[int]bool // `_TypeA_to_sumtype_TypeB()` fns that have been generated
+	trace_fn_definitions    []string
+	json_types              []ast.Type // to avoid json gen duplicates
+	pcs                     []ProfileCounterMeta // -prof profile counter fn_names => fn counter name
+	hotcode_fn_names        []string
+	hotcode_fpaths          []string
+	embedded_files          []ast.EmbeddedFile
+	sql_i                   int
+	sql_stmt_name           string
+	sql_bind_name           string
+	sql_idents              []string
+	sql_idents_types        []ast.Type
+	sql_left_type           ast.Type
+	sql_table_name          string
+	sql_fkey                string
+	sql_parent_id           string
+	sql_side                SqlExprSide // left or right, to distinguish idents in `name == name`
+	strs_to_free0           []string    // strings.Builder
 	// strs_to_free          []string // strings.Builder
 	// tmp_arg_vars_to_free  []string
 	// autofree_pregen       map[string]string

